plugins/org.eclipse.n4js/src/org/eclipse/n4js/typesystem/constraints/Reducer.java

```diff
@@ -627,11 +627,40 @@ private boolean reduceParameterizedTypeRef(ParameterizedTypeRef left, Parameteri
 						throw new UnsupportedOperationException("unsupported subtype of TypeArgument: "
 								+ leftArg.getClass().getName());
 					}
-					wasAdded |= reduce(leftArg, leftParamSubst, variance.mult(INV));
+					final TypeVariable leftTypeVar = (left.getDeclaredType() != null
+							&& left.getDeclaredType().getTypeVars().size() > 0)
+									? left.getDeclaredType().getTypeVars().get(0) : null;
+					final TypeVariable rightTypeVar = (right.getDeclaredType() != null
+							&& right.getDeclaredType().getTypeVars().size() >= idx)
+									? right.getDeclaredType().getTypeVars().get(idx) : null;
+
+					boolean caseConvered = false;
+					if (variance == CO) {
```

Comment:
can this ever happen? (see normalization in lines 582-588, above)

Reply:
Done. Remove this case.